src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -257,33 +342,47 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
+            var unformattedInterfaceDocument = await CodeGenerator.AddNamespaceOrTypeDeclarationAsync(
                 interfaceDocument.Project.Solution,
                 interfaceDocumentSemanticModel.GetEnclosingNamespace(0, cancellationToken),
                 extractedInterfaceSymbol.GenerateRootNamespaceOrType(namespaceParts.ToArray()),
                 options: new CodeGenerationOptions(interfaceDocumentSemanticModel.SyntaxTree.GetLocation(new TextSpan())),
-                cancellationToken: cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
+                cancellationToken: cancellationToken).ConfigureAwait(false);
 
             return unformattedInterfaceDocument;
         }
 
-        private static Solution GetSolutionWithFormattedInterfaceDocument(Document unformattedInterfaceDocument, CancellationToken cancellationToken)
+        private static async Task<Solution> GetSolutionWithFormattedInterfaceDocument(Document unformattedInterfaceDocument, CancellationToken cancellationToken)
         {
             Solution solutionWithInterfaceDocument;
             var formattedRoot = Formatter.Format(unformattedInterfaceDocument.GetSyntaxRootSynchronously(cancellationToken),
                 unformattedInterfaceDocument.Project.Solution.Workspace, cancellationToken: cancellationToken);
             var rootToSimplify = formattedRoot.WithAdditionalAnnotations(Simplifier.Annotation);
-            var finalInterfaceDocument = Simplifier.ReduceAsync(unformattedInterfaceDocument.WithSyntaxRoot(rootToSimplify), cancellationToken: cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
+            var finalInterfaceDocument = await Simplifier.ReduceAsync(unformattedInterfaceDocument.WithSyntaxRoot(rootToSimplify), cancellationToken: cancellationToken).ConfigureAwait(false);
 
             solutionWithInterfaceDocument = finalInterfaceDocument.Project.Solution;
             return solutionWithInterfaceDocument;
         }
 
-        private Solution GetSolutionWithOriginalTypeUpdated(
-            Solution solutionWithFormattedInterfaceDocument,
+        private async Task<Solution> GetFormattedSolutionAsync(Solution unformattedSolution, IEnumerable<DocumentId> documentIds, CancellationToken cancellationToken)
+        {
+            var formattedSolution = unformattedSolution;
+            foreach (var documentId in documentIds)
+            {
+                var document = formattedSolution.GetDocument(documentId);
+                var formattedDocument = await Formatter.FormatAsync(document, cancellationToken: cancellationToken).ConfigureAwait(false);
+                var simplifiedDocument = await Simplifier.ReduceAsync(formattedDocument, cancellationToken: cancellationToken).ConfigureAwait(false);
+                formattedSolution = simplifiedDocument.Project.Solution;
+            }
+
+            return formattedSolution;
```

Comment:
if you want to be cute, do this all in parallel :)

Reply:
I would personally prefer that the issue not be filed.  The repo is so cluttered at this point with ideas that simply will never be done.  So we should either do it, or not. I'm 100% ok with not doing it. But i def would like to get a handle on the enormous amount of chaff that is filling the issue tracker :-/

Tagging @sharwell for thoughts on this.

Reply:
#31378